aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -1986,6 +2113,200 @@ internal class RealAWSCognitoAuthPlugin(
+            }
+        }
+    }
+
+    override fun verifyTOTPSetup(
+        code: String,
+        onSuccess: Action,
+        onError: Consumer<AuthException>
+    ) {
+        verifyTotp(code, null, onSuccess, onError)
+    }
+
+    override fun verifyTOTPSetup(
+        code: String,
+        options: AuthVerifyTOTPSetupOptions,
+        onSuccess: Action,
+        onError: Consumer<AuthException>
+    ) {
+        val cognitoOptions = options as? AWSAuthVerifyTOTPSetupOptions
+        verifyTotp(code, cognitoOptions?.friendlyDeviceName, onSuccess, onError)
+    }
+
+    fun fetchMFAPreference(
+        onSuccess: Consumer<UserMFAPreference>,
+        onError: Consumer<AuthException>
+    ) {
+        authStateMachine.getCurrentState { authState ->
+            when (authState.authNState) {
+                is AuthenticationState.SignedIn -> {
+                    GlobalScope.launch {
+                        try {
+                            val accessToken = getSession().userPoolTokensResult.value?.accessToken
+                            accessToken?.let { token ->
+                                authEnvironment.cognitoAuthService
+                                    .cognitoIdentityProviderClient?.getUser {
+                                        this.accessToken = token
+                                    }?.also { response ->
+                                        var enabledSet: MutableSet<MFAType>? = null
+                                        var preferred: MFAType? = null
+                                        if (!response.userMfaSettingList.isNullOrEmpty()) {
+                                            enabledSet = mutableSetOf<MFAType>()
+                                            response.userMfaSettingList?.forEach { mfaType ->
+                                                enabledSet.add(MFAType.toMFAType(mfaType))
+                                            }
+                                        }
+                                        response.preferredMfaSetting?.let { preferredMFA ->
+                                            preferred = MFAType.toMFAType(preferredMFA)
+                                        }
+                                        onSuccess.accept(UserMFAPreference(enabledSet, preferred))
+                                    }
+                            } ?: onError.accept(SignedOutException())
+                        } catch (error: Exception) {
+                            onError.accept(
+                                CognitoAuthExceptionConverter.lookup(
+                                    error,
+                                    "Cannot update the MFA preferences"
+                                )
+                            )
+                        }
+                    }
+                }
+
+                else -> onError.accept(InvalidStateException())
+            }
+        }
+    }
+
+    fun updateMFAPreference(
```

Comment:
No state machine?

Reply:
yeah its not required as that will be extra overhead. The api is called directly.